softlayer/vm/softlayer_creator.go

```diff
@@ -41,9 +41,29 @@ func NewSoftLayerCreator(softLayerClient sl.Client, agentEnvServiceFactory Agent
 	}
 }
 
+// get the time stample of action occurs to name the vm hostname
+func (c SoftLayerCreator) getTimeStample(now time.Time) string {
+	return now.Format("20060102-030405-") + strconv.Itoa(int(now.UnixNano()/1e6-now.Unix()*1e3))
+}
+
+func (c SoftLayerCreator) checkValid(t *sldatatypes.SoftLayer_Virtual_Guest_Template) *sldatatypes.SoftLayer_Virtual_Guest_Template {
+	//check domain name configure or not
```

Comment:
Again, no need for comment since the code is clear. The comment distracts and adds nothing.